vendor/plist_to_html/plist_to_html/PlistToHtml.py

```diff
@@ -0,0 +1,206 @@
+                           'col':  event.location['col'],
+                           'file': event.location['file'],
+                           'msg':  event.message,
+                           'step': index + 1})
+        reports.append(report)
+
+    return {'files': file_sources,
+            'reports': reports}
+
+
+def plist_to_html(file_path, output_path, html_builder):
+    """
+    Prints the results in the given file to HTML file.
+    """
+    if not file_path.endswith(".plist"):
+        print("Skipping input file '" + file_path + "' as it is not a plist.")
+        return
+
+    print("Parsing input file '" + file_path + "'")
+
+    try:
+        plist = plistlib.readPlist(file_path)
+
+        report_data = get_report_data_from_plist(plist)
+
+        if report_data is None or not len(report_data['reports']):
+            print('No report data in ' + file_path + ' file.')
+            return
+
+        html_filename = os.path.basename(file_path) + '.html'
+        html_output_path = os.path.join(output_path, html_filename)
+        html_builder.create(html_output_path, report_data)
+
+        print('Html file was generated: ' + html_output_path)
+    except ExpatError as err:
+        print('Failed to process plist file: ' + file_path +
+              ' wrong file format?', err)
+    except AttributeError as ex:
+        print('Failed to get important report data from plist.', ex)
+    except IndexError as iex:
+        print('Indexing error during processing plist file ' + file_path, iex)
+    except Exception as ex:
+        print('Error during processing reports from the plist file: ' +
+              file_path, ex)
+
+
+def parse(input_path, output_path, layout_dir, clean=False):
+    files = []
+    input_path = os.path.abspath(input_path)
+
+    output_dir = os.path.abspath(output_path)
+    if clean and os.path.isdir(output_dir):
+        print("Previous analysis results in '{0}' have been removed, "
+              "overwriting with current result".format(output_dir))
+        shutil.rmtree(output_path)
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    if os.path.isfile(input_path):
+        files.append(input_path)
+    elif os.path.isdir(input_path):
+        _, _, file_names = next(os.walk(input_path), ([], [], []))
+        files = [os.path.join(input_path, file_name) for file_name
+                 in file_names]
+
+    html_builder = HtmlBuilder(layout_dir)
+    for file_path in files:
+        plist_to_html(file_path, output_path, html_builder)
+
+
+def __add_arguments_to_parser(parser):
+    parser.add_argument('--input',
+                        type=str,
+                        nargs='*',
+                        metavar='file/folder',
+                        help="The plist files and/or folders containing "
+                             "analysis results which should be parsed.")
+
+    parser.add_argument('-o', '--output',
```

Comment:
Shouldn't at least one of these arguments, either the output or the input(s) be a **positional** argument instead of an optional one?

Reply:
I made the `input` argument to be a positional argument.